lib/active_merchant/billing/gateways/paypal.rb

```diff
@@ -98,6 +105,19 @@ def add_descriptors(xml, options)
         xml.tag! 'n2:SoftDescriptorCity', options[:soft_descriptor_city] unless options[:soft_descriptor_city].blank?
       end
 
+      def add_three_d_secure(xml, options)
+        three_d_secure = options[:three_d_secure]
+        xml.tag! 'ThreeDSecureRequest' do
+          xml.tag! 'MpiVendor3ds', 'Y'
+          xml.tag! 'AuthStatus3ds', three_d_secure[:trans_status] unless three_d_secure[:trans_status].blank?
```

Comment:
Discussed with Filipe: `trans_status` is [required by Paypal](https://developer.paypal.com/docs/classic/paypal-payments-pro/integration-guide/3d-secure/#integration-steps).